params/config_extra.go

```diff
@@ -40,6 +40,24 @@ type AvalancheContext struct {
 // code in place of their Ethereum counterparts. The original Ethereum names
 // should be restored for maintainability.
 func (c *ChainConfig) SetEthUpgrades() {
+	if c.ChainID != nil && AvalancheFujiChainID.Cmp(c.ChainID) == 0 {
```

Comment:
Can we rather check with network ID from chain context? This should be good but in practice there can be local genesis confs using prod chain IDs (even they should not be doing that).

Reply:
I made this modification, however we should explicitly discourage use of coreth other than fuji/mainnet.
Everyone else will be better off using subnet-evm, as we have already deviated from preserving hypothetical historical processing (eg, if some local network used previous opcodes which we removed they will no longer function).

Reply:
Also since we don't store the networkID, it creates the need to use the stored config with the new network ID, so I'm not very sure what we should do

Reply:
hmm yea sorry I did not notice that. I think we can use the chainID only for block based eth upgrades if using network ID seems weird in genesis.